samples/bluetooth/hci_uart/src/main.c

```diff
@@ -281,47 +283,48 @@ static int h4_send(struct net_buf *buf)
 #if defined(CONFIG_BT_CTLR_ASSERT_HANDLER)
 void bt_ctlr_assert_handle(char *file, uint32_t line)
 {
-	uint32_t len = 0U, pos = 0U;
-
 	/* Disable interrupts, this is unrecoverable */
 	(void)irq_lock();
 
-	uart_irq_rx_disable(hci_uart_dev);
-	uart_irq_tx_disable(hci_uart_dev);
+#if defined(CONFIG_BT_HCI_VS_FATAL_ERROR)
+	/* Prepare vendor specific HCI debug event */
+	struct net_buf *buf;
 
-	if (file) {
-		while (file[len] != '\0') {
-			if (file[len] == '/') {
-				pos = len + 1;
-			}
-			len++;
-		}
-		file += pos;
-		len -= pos;
-	}
+	buf = hci_vs_err_assert(file, line);
```

Comment:
Won't this treat file and line as a file name and line number instead of a fixed string and a PC value?